INSTALL

```diff
@@ -30,12 +30,25 @@
 ################################################################################
 
 Installation Instructions for STACCATO
-$> make
-$> make test (edit the CUDD and DSD library paths)
 
+To change the static/shared library type, use the make parameter:
+BUILD_TYPE=shared or BUILD_TYPE=static (static by default).
+You must specify the path to the CUDD sources directory with:
+CUDD_PATH=example/path/to/cudd.
+
+Example for building shared library:
+$> make BUILD_TYPE=shared CUDD_PATH=example/path/to/cudd
```

Comment:
I've tried to do this, and I've received the following error:
```
$ make BUILD_TYPE=shared CUDD_PATH=/home/ssedai/sources/cudd
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd/include   -c -o src/DSDDecompose.o src/DSDDecompose.c
In file included from src/DSDDecompose.h:36,
                 from src/DSDDecompose.c:33:
src/DSD.h:41:10: fatal error: util.h: No such file or directory
   41 | #include "util.h"
      |          ^~~~~~~~
compilation terminated.
make: *** [<builtin>: src/DSDDecompose.o] Error 1
```
When I've tried to find the missing `util.h` in my CUDD sources, I've found no such file there. This file exists in https://github.com/ivmai/cudd repository neither. 
Where should I get CUDD to make this building successful?

Reply:
Regarding the error with the assembly. The file itself is in util/utils.h. It didn't build because I forgot that I ran the script in cudd, which created the include folder with all header files.

Reply:
So, what to do? Are weed need to create our own CUDD fork with this enhancement also?

Reply:
No, I'll just explicitly write the paths to the header files when building.

Reply:
Finally, `INSTALL` instruction should be updated for other users to use it. Now I can't install staccato library with it and here is a problem.

Reply:
I did, also fix test target.

Reply:
Trying to build shared library, now I get the following:
```
$ make BUILD_TYPE=shared CUDD_INCLUDE=/home/ssedai/sources/cudd/
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDDecompose.o src/DSDDecompose.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDInterface.o src/DSDInterface.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDManager.o src/DSDManager.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDManipulations.o src/DSDManipulations.c
src/DSDManipulations.c: In function ‘void __Decomposition_Print(DSDNode*)’:
src/DSDManipulations.c:78:14: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 2 has type ‘DSDNode*’ [-Wformat=]
   78 |     printf("%x\n", dsd_node);
      |             ~^     ~~~~~~~~
      |              |     |
      |              |     DSDNode*
      |              unsigned int
src/DSDManipulations.c:81:16: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 2 has type ‘DdNode*’ [-Wformat=]
   81 |     printf("\t%x\t\t%x\n\n", __Get_Symbolic_Decomposition(dsd_node), __Get_BDD(dsd_node));
      |               ~^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                |                                         |
      |                unsigned int                              DdNode*
src/DSDManipulations.c:81:22: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘DdNode*’ [-Wformat=]
   81 |     printf("\t%x\t\t%x\n\n", __Get_Symbolic_Decomposition(dsd_node), __Get_BDD(dsd_node));
      |                     ~^                                               ~~~~~~~~~~~~~~~~~~~
      |                      |                                                        |
      |                      unsigned int                                             DdNode*
src/DSDManipulations.c:92:22: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 2 has type ‘DSDNode*’ [-Wformat=]
   92 |         printf("\t\t%x\t\t%x\t\t%x\n", iter->decomposition, __Get_Symbolic_Decomposition(iter->decomposition), __Get_BDD(iter->decomposition));
      |                     ~^                 ~~~~~~~~~~~~~~~~~~~
      |                      |                       |
      |                      unsigned int            DSDNode*
src/DSDManipulations.c:92:28: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘DdNode*’ [-Wformat=]
   92 |         printf("\t\t%x\t\t%x\t\t%x\n", iter->decomposition, __Get_Symbolic_Decomposition(iter->decomposition), __Get_BDD(iter->decomposition));
      |                           ~^                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            |                                                            |
      |                            unsigned int                                                 DdNode*
src/DSDManipulations.c:92:34: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 4 has type ‘DdNode*’ [-Wformat=]
   92 |         printf("\t\t%x\t\t%x\t\t%x\n", iter->decomposition, __Get_Symbolic_Decomposition(iter->decomposition), __Get_BDD(iter->decomposition));
      |                                 ~^                                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                  |                                                                                      |
      |                                  unsigned int                                                                           DdNode*
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDNewDecompose.o src/DSDNewDecompose.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDOrDecompose.o src/DSDOrDecompose.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDPrimeDecompose.o src/DSDPrimeDecompose.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDUtilities.o src/DSDUtilities.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/DSDXorDecompose.o src/DSDXorDecompose.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/fixheap.o src/fixheap.c
g++ -c -O3 -funroll-all-loops -fPIC -I/home/ssedai/sources/cudd//util -I/home/ssedai/sources/cudd//cudd -I/home/ssedai/sources/cudd/ -I/home/ssedai/sources/cudd//st -I/home/ssedai/sources/cudd//mtr -I/home/ssedai/sources/cudd//epd   -c -o src/sample.o src/sample.c
make: *** No rule to make target '/usr/local/lib/libcudd.so', needed by 'libSTACCATO.so'.  Stop.
```
Please, check this instruction on clean machine!

Reply:
I've updated dependences in install.

Reply:
Yeah, it works now. Please move "Dependencies" section before "Installation Instructions", because it is necessary to build CUDD with `--enable-shared` before STACCATO.

Reply:
Did